scripts/scil_estimate_bundles_diameter.py

```diff
@@ -0,0 +1,348 @@
+    <5 is not enough, high risk of bad fit
+    >50 is too much, high risk of bad fit
+- bundles that are close to a tube
+    without major fanning in a single axis
+    fanning is in 2 directions (uniform dispersion) good approximation
+
+The scripts print a JSON file with mean/std to be compatible with tractometry.
+WARNING: STD is in fact an ERROR measure from the fit and NOT an STD.
+
+Since the estimation and fit quality is not always intuitive for some bundles
+and the tube with varying diameter is not easy to color/visualize,
+the script comes with its own VTK rendering to allow exploration of the data.
+(optional).
+"""
+
+import argparse
+import json
+import os
+
+from dipy.io.utils import is_header_compatible
+from fury import window, actor
+import matplotlib.pyplot as plt
+import nibabel as nib
+import numpy as np
+from scipy.linalg import svd
+from scipy.ndimage import map_coordinates
+from scipy.ndimage.filters import gaussian_filter
+import vtk
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_reference_arg,
+                             add_json_args,
+                             assert_inputs_exist,
+                             parser_color_type)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_bundles', nargs='+',
+                   help='List of tractography files supported by nibabel.')
+    p.add_argument('in_labels', nargs='+',
+                   help='List of labels maps that matches the bundles.')
+
+    p.add_argument('--fitting_func', choices=['lin_up', 'lin_down', 'exp',
+                                              'inv', 'log'],
+                   help='Function to weigh points using their distance.')
+
+    p2 = p.add_argument_group(title='Visualization options')
+    p2.add_argument('--show_rendering', action='store_true',
+                    help='Display VTK window.')
+    p2.add_argument('--wireframe', action='store_true',
+                    help='Use wireframe for the tube rendering.')
+    p2.add_argument('--error_coloring', action='store_true',
+                    help='Use the fitting error to color the tube.')
+    p2.add_argument('--width', type=float, default=0.2,
+                    help='Width of tubes or lines representing streamlines'
+                    '\n[Default: %(default)s]')
+    p2.add_argument('--opacity', type=float, default=0.1,
+                    help='Opacity for the streamlines rendered with the tube.'
+                    '\n[Default: %(default)s]')
+    p2.add_argument('--background', metavar=('R', 'G', 'B'), nargs=3,
+                    default=[0, 0, 0], type=parser_color_type,
+                    help='RBG values [0, 255] of the color of the background.'
+                    '\n[Default: %(default)s]')
```

Comment:
Can you add a save png option? It would help if you want just to script and output a bunch of PNGs.

Reply:
This adds quite a bit of horrible code. Since it can be used on any bundles it would require taking pictures from left/right/up/down/front/back and estimating the distances (which would chance from bundle to bundle, and subject to subject).

Is it really worth it? I personally saw the display as a debugging tool and making nice images for papers.